components/setting_item_max.jsx

```diff
@@ -132,7 +132,7 @@ export default class SettingItemMax extends React.PureComponent {
         if (this.props.shiftEnter && e.keyCode === Constants.KeyCodes.ENTER && e.shiftKey) {
             return;
         }
-        if (isKeyPressed(e, Constants.KeyCodes.ENTER) && this.props.submit && e.target.tagName !== 'SELECT' && e.target.parentElement && e.target.parentElement.className !== 'react-select__input') {
+        if (isKeyPressed(e, Constants.KeyCodes.ENTER) && this.props.submit && e.target.tagName !== 'SELECT' && e.target.parentElement && e.target.parentElement.className !== 'react-select__input' && !e.target.classList.contains('btn-cancel')) {
```

Comment:
We could use `e.target.id` an compare it with `cancelSetting`. Any reason why we need the className?

Reply:
Yup, we can do that too (for this case). With id's it's just that only one element can be present on the page, but with this class, we can just give it to other buttons that we want to prevent a submit behaviour on.

Reply:
Aah, we should not be propagating these events up the DOM chain unless required. I don't know if we use them higher up in the chain. 

> but with this class, we can just give it to other buttons that we want to prevent a submit behaviour on.

We should not be using this even for other buttons to even exclude it IMO. This component should only be listning and responding to the the click originated from within

I didn't realise we use this component in multiple places at once. Having the same ID for all of these instances is another odd thing as it violates the HTMl.

Requested change because thought we wouldn't need it. Never mind it. Class works anyway for the behaviour.